src/internal/packager/images/pull.go

```diff
@@ -168,3 +191,63 @@ func (i *ImgConfig) PullImage(src string, spinner *message.Spinner) (img v1.Imag
 
 	return img, nil
 }
+
+// IndexJSON represents the index.json file in an OCI layout.
+type IndexJSON struct {
+	SchemaVersion int `json:"schemaVersion"`
+	Manifests     []struct {
+		MediaType   string            `json:"mediaType"`
+		Size        int               `json:"size"`
+		Digest      string            `json:"digest"`
+		Annotations map[string]string `json:"annotations"`
+	} `json:"manifests"`
+}
+
+// addImageNameAnnotation adds an annotation to the index.json file so that the deploying code can figure out what the image tag <-> digest shasum will be.
+func addImageNameAnnotation(ociPath string, digestToTag map[string]string) error {
+	indexPath := filepath.Join(ociPath, "index.json")
+
+	// Add an 'org.opencontainers.image.base.name' annotation so we can figure out what the image tag/digest shasum will be during deploy time
```

Comment:
The intended purpose of `org.opencontainers.image.base.name` is a base image, like a Dockerfile FROM. This is sort of a misuse of this annotation. The digest here _is_ the same image the base.

I think `org.opencontainers.image.ref.name` is more appropriate? The [image layout specification](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) says it's usually only a tag, but also says there is "no semantic restriction".

Reply:
@YrrepNoj FYI